src/content/docs/ci-insights/cypress.mdx

```diff
@@ -0,0 +1,85 @@
+---
+title: Cypress Integration with CI Insights
+description: Report your test results from Cypress to CI Insights
+---
+
+import cypressLogo from "../../images/ci-insights/cypress/logo.svg"
+import IntegrationLogo from "../../../components/IntegrationLogo.astro"
+
+<IntegrationLogo src={cypressLogo} alt="Cypress logo"/>
+
+This guide shows how to configure [Cypress](https://www.cypress.io/) to produce
+JUnit reports and upload them to CI Insights within a GitHub Actions workflow.
+
+## 1. Generate a JUnit Report with Cypress
+
+By default, Cypress doesn't generate JUnit-format test results. You need to
+enable it in your `cypress.config.ts` file.
+
+This is [documenteded on Cypress
+website](https://docs.cypress.io/app/tooling/reporters).
+
+The best way to do this is to leverage `cypress-multi-reporters` so you can
+both have JUnit generated while keeping the standard output.
+
+1. Add `cypress-multi-reporters` to your package file, e.g.,
+   `npm install --save-dev cypress-multi-reporters mocha-junit-reporter`.
+
+2. Create a `reporter-config.json` at the top of your repository with this
+   configuration:
+   ```yaml
+   {
+     "reporterEnabled": "spec, mocha-junit-reporter",
+     "mochaJunitReporterReporterOptions": {
+       "mochaFile": "cypress/results/junit-[hash].xml"
+     }
+   }
+   ```
+
+## 2. Update Your GitHub Actions Workflow
+
+Once you have the JUnit file produced by Cypress, add a step to upload these
+results to **CI Insights** via the `mergifyio/gha-mergify-ci` action.
+
+In the workflow file where Cypress is launched, after running `npm run
+test:cypress` (or similar), include a step like:
+
+```yaml
+- name: Mergify CI Upload
+  if: success() || failure()
+  uses: mergifyio/gha-mergify-ci@v4
+  with:
+    token: ${{ secrets.MERGIFY_TOKEN }}
+    report_paths: cypress/results/junit*.xml
+```
+
+Key Points:
+
+- `if: success() || failure()`: Runs the upload step even if tests
+  fail, ensuring **CI Insights** has the full report.
+
+- `report_paths: src/junit.xml`: Points to where your JUnit file is located.
+  Make sure it matches the `outputFile` path you set in Cypress (e.g.,
+  `./cypress/results/junit-*.xml`).
+
+## 3. Verify and Review in CI Insights
+
+After pushing these changes:
+1. Your GitHub Actions workflow will run tests with Cypress.
+2. Cypress generates JUnit files.
+3. The Mergify CI action uploads that file to CI Insights.
+
+You can then see your test results, including failures and flaky tests,
+directly in the CI Insights dashboard.
+
+## Troubleshooting Tips
+
+- File Paths: Double-check that `mochaFile` in Cypress matches the path used in
+  `report_paths`.
+
+- Permissions: Make sure the `MERGIFY_TOKEN` is valid and setup in your GitHub
+  Actions secrets [as explained in the
+  docs]((/ci-insights#enabling-ci-insights-for-github)).
```

Comment:
This should also be mentioned in the `Key Points` in step 2 imo

Reply:
Key points explain what asked the user to write; this helps debugging. Instructions to set things up are already provided in the main page.